src/Eloquent/Model.php

```diff
@@ -222,13 +228,6 @@ public function attributesToArray()
             }
         }
 
-        // Convert dot-notation dates.
```

Comment:
This was discussed and the decision was to keep dot notation support, anyways I'd prefer removing it honestly but some discussion is needed. @Smolevich what do you think?

Reply:
This is still supported, in fact this PR expands support for dot notation. Since dates are now handled via a cast, they will be covered by `addCastAttributesToArray`. This method was added to this class to support dot-notation for any cast instead of only magically casting date fields. In the future, it will be possible to have something like this:
```php
protected $casts = [
    'entry.date' => 'datetime',
    'entry.list' => 'array',
];
```